Maven Plugin tests fail on Windows

The tests of the maven plugin fail on windows.
```
[ERROR] Failures:
[ERROR]   SchemaGeneratorMojoTest.testFileNamePattern:203 Generated schema for TestClassA is not equal to the expected reference. ==> expected: <true> but was: <false>
[ERROR]   SchemaGeneratorMojoTest.testGeneration:74 Generated schema for DefaultConfig is not equal to the expected reference. ==> expected: <true> but was: <false>
[ERROR]   SchemaGeneratorMojoTest.testGeneration:74 Generated schema for SchemaVersion is not equal to the expected reference. ==> expected: <true> but was: <false>
[ERROR]   SchemaGeneratorMojoTest.testGeneration:74 Generated schema for JacksonModule is not equal to the expected reference. ==> expected: <true> but was: <false>
[ERROR]   SchemaGeneratorMojoTest.testGeneration:74 Generated schema for JavaxValidationModule is not equal to the expected reference. ==> expected: <true> but was: <false>
[ERROR]   SchemaGeneratorMojoTest.testGeneration:74 Generated schema for JakartaValidationModule is not equal to the expected reference. ==> expected: <true> but was: <false>
[ERROR]   SchemaGeneratorMojoTest.testGeneration:74 Generated schema for Swagger15Module is not equal to the expected reference. ==> expected: <true> but was: <false>
[ERROR]   SchemaGeneratorMojoTest.testGeneration:74 Generated schema for Swagger2Module is not equal to the expected reference. ==> expected: <true> but was: <false>
[ERROR]   SchemaGeneratorMojoTest.testGeneration:74 Generated schema for Complete is not equal to the expected reference. ==> expected: <true> but was: <false>
[ERROR]   SchemaGeneratorMojoTest.testPackageName:165 Generated schema for TestClassA is not equal to the expected reference. ==> expected: <true> but was: <false>
[ERROR]   SchemaGeneratorMojoTest.testTwoClasses:129 Generated schema for TestClassA is not equal to the expected reference. ==> expected: <true> but was: <false>
```

This is most likely the cause, because the default Jackson pretty printer uses `\r\n` on windows, but the reference test files only have `\n`. See [here](https://github.com/FasterXML/jackson-databind/issues/585)

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -9,6 +9,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 #### Added
 - support `@Schema.anyOf` and `@Schema.oneOf` on fields/methods
 
+### `jsonschema-maven-plugin`
+#### Added
+- support `<failIfNoClassesMatch>false</failIfNoClassesMatch>` parameter, in order to continue build even if no class matches the defined pattern(s)
+
 ## [4.25.0] - 2022-06-24
 ### `jsonschema-generator`
 #### Added

--- a/jsonschema-generator-parent/pom.xml
+++ b/jsonschema-generator-parent/pom.xml
@@ -90,6 +90,13 @@
                 <role>Provided PR #254 (introducing Option.FLATTENED_SUPPLIERS)</role>
             </roles>
         </contributor>
+        <contributor>
+            <name>Florian Beutel</name>
+            <url>https://github.com/Emmeral</url>
+            <roles>
+                <role>Provided PR #273 (introducing failIfNoClassesMatch parameter on maven plugin)</role>
+            </roles>
+        </contributor>
     </contributors>
 
     <properties>

--- a/jsonschema-maven-plugin/src/main/java/com/github/victools/jsonschema/plugin/maven/SchemaGeneratorMojo.java
+++ b/jsonschema-maven-plugin/src/main/java/com/github/victools/jsonschema/plugin/maven/SchemaGeneratorMojo.java
@@ -210,11 +210,10 @@ public class SchemaGeneratorMojo extends AbstractMojo {
             if (this.excludeClassNames != null && this.excludeClassNames.length > 0) {
                 message.append(" that wasn't excluded");
             }
-            if (failIfNoClassesMatch) {
+            if (this.failIfNoClassesMatch) {
                 throw new MojoExecutionException(message.toString());
-            } else {
-                getLog().warn(message.toString());
             }
+            this.getLog().warn(message.toString());
         }
     }
 

--- a/jsonschema-maven-plugin/src/test/java/com/github/victools/jsonschema/plugin/maven/SchemaGeneratorMojoTest.java
+++ b/jsonschema-maven-plugin/src/test/java/com/github/victools/jsonschema/plugin/maven/SchemaGeneratorMojoTest.java
@@ -90,6 +90,18 @@ public class SchemaGeneratorMojoTest extends AbstractMojoTestCase{
     }
 
     /**
+     * Unit test that will generate from a maven pom file fragment and expect no exception being thrown although no classes matches pattern.
+     *
+     * @throws Exception In case something goes wrong
+     */
+    @Test
+    public void testDontFailIfNoClassesMatch() throws Exception {
+        File testCaseLocation = new File("src/test/resources/error-test-cases");
+        this.executePom(new File(testCaseLocation, "ClassNotFound-dontFailIfNoClassesMatch-pom.xml"));
+        // no error thrown, i.e., build proceeds even without any schema having been generated
+    }
+
+    /**
      * Unit test that will generate from a maven pom file fragment and expect a ComponentConfigurationException.
      *
      * @param testCaseName Name of the test case and file name prefix of the example {@code pom.xml}

--- a/jsonschema-maven-plugin/src/test/java/com/github/victools/jsonschema/plugin/maven/SchemaGeneratorMojoTest.java
+++ b/jsonschema-maven-plugin/src/test/java/com/github/victools/jsonschema/plugin/maven/SchemaGeneratorMojoTest.java
@@ -72,7 +72,7 @@ public class SchemaGeneratorMojoTest extends AbstractMojoTestCase {
         resultFile.deleteOnExit();
 
         // Validate that is the same as the reference
-        File referenceFile = new File(testCaseLocation + "/" + testCaseName + "-reference.json");
+        File referenceFile = new File(testCaseLocation, testCaseName + "-reference.json");
         Assertions.assertTrue(referenceFile.exists());
         Assertions.assertTrue(FileUtils.contentEqualsIgnoreEOL(resultFile, referenceFile, CHARSET_NAME),
                 "Generated schema for " + testCaseName + " is not equal to the expected reference.");
@@ -139,12 +139,12 @@ public class SchemaGeneratorMojoTest extends AbstractMojoTestCase {
         resultFileB.deleteOnExit();
 
         // Validate that they are the same as the reference
-        File referenceFileA = new File(testCaseLocation + "/" + "TestClassA-reference.json");
+        File referenceFileA = new File(testCaseLocation, "TestClassA-reference.json");
         Assertions.assertTrue(referenceFileA.exists());
         Assertions.assertTrue(FileUtils.contentEqualsIgnoreEOL(resultFileA, referenceFileA, CHARSET_NAME),
                 "Generated schema for TestClassA is not equal to the expected reference.");
 
-        File referenceFileB = new File(testCaseLocation + "/" + "TestClassB-reference.json");
+        File referenceFileB = new File(testCaseLocation, "TestClassB-reference.json");
         Assertions.assertTrue(referenceFileB.exists());
         Assertions.assertTrue(FileUtils.contentEqualsIgnoreEOL(resultFileB, referenceFileB, CHARSET_NAME),
                 "Generated schema for TestClassB is not equal to the expected reference.");
@@ -175,17 +175,17 @@ public class SchemaGeneratorMojoTest extends AbstractMojoTestCase {
         resultFileC.deleteOnExit();
 
         // Validate that they are the same as the reference
-        File referenceFileA = new File(testCaseLocation + "/" + "TestClassA-reference.json");
+        File referenceFileA = new File(testCaseLocation, "TestClassA-reference.json");
         Assertions.assertTrue(referenceFileA.exists());
         Assertions.assertTrue(FileUtils.contentEqualsIgnoreEOL(resultFileA, referenceFileA, CHARSET_NAME),
                 "Generated schema for TestClassA is not equal to the expected reference.");
 
-        File referenceFileB = new File(testCaseLocation + "/" + "TestClassB-reference.json");
+        File referenceFileB = new File(testCaseLocation, "TestClassB-reference.json");
         Assertions.assertTrue(referenceFileB.exists());
         Assertions.assertTrue(FileUtils.contentEqualsIgnoreEOL(resultFileB, referenceFileB, CHARSET_NAME),
                 "Generated schema for TestClassB is not equal to the expected reference.");
 
-        File referenceFileC = new File(testCaseLocation + "/" + "TestClassC-reference.json");
+        File referenceFileC = new File(testCaseLocation, "TestClassC-reference.json");
         Assertions.assertTrue(referenceFileC.exists());
         Assertions.assertTrue(FileUtils.contentEqualsIgnoreEOL(resultFileC, referenceFileC, CHARSET_NAME),
                 "Generated schema for TestClassC is not equal to the expected reference.");
@@ -213,12 +213,12 @@ public class SchemaGeneratorMojoTest extends AbstractMojoTestCase {
         resultFileB.deleteOnExit();
 
         // Validate that they are the same as the reference
-        File referenceFileA = new File(testCaseLocation + "/" + "TestClassA-reference.json");
+        File referenceFileA = new File(testCaseLocation, "TestClassA-reference.json");
         Assertions.assertTrue(referenceFileA.exists());
         Assertions.assertTrue(FileUtils.contentEqualsIgnoreEOL(resultFileA, referenceFileA, CHARSET_NAME),
                 "Generated schema for TestClassA is not equal to the expected reference.");
 
-        File referenceFileB = new File(testCaseLocation + "/" + "TestClassB-reference.json");
+        File referenceFileB = new File(testCaseLocation, "TestClassB-reference.json");
         Assertions.assertTrue(referenceFileB.exists());
         Assertions.assertTrue(FileUtils.contentEqualsIgnoreEOL(resultFileB, referenceFileB, CHARSET_NAME),
                 "Generated schema for TestClassB is not equal to the expected reference.");

--- a/jsonschema-maven-plugin/src/test/java/com/github/victools/jsonschema/plugin/maven/SchemaGeneratorMojoTest.java
+++ b/jsonschema-maven-plugin/src/test/java/com/github/victools/jsonschema/plugin/maven/SchemaGeneratorMojoTest.java
@@ -18,6 +18,7 @@ package com.github.victools.jsonschema.plugin.maven;
 
 import java.io.File;
 import java.io.FileReader;
+import java.nio.charset.StandardCharsets;
 import org.apache.commons.io.FileUtils;
 import org.apache.maven.plugin.MojoExecutionException;
 import org.apache.maven.plugin.testing.AbstractMojoTestCase;
@@ -32,7 +33,9 @@ import org.junit.jupiter.api.Test;
 import org.junit.jupiter.params.ParameterizedTest;
 import org.junit.jupiter.params.provider.ValueSource;
 
-public class SchemaGeneratorMojoTest extends AbstractMojoTestCase{
+public class SchemaGeneratorMojoTest extends AbstractMojoTestCase {
+
+    private static final String CHARSET_NAME = StandardCharsets.UTF_8.name();
 
     @BeforeEach
     @Override
@@ -64,14 +67,14 @@ public class SchemaGeneratorMojoTest extends AbstractMojoTestCase{
         executePom(new File(testCaseLocation, testCaseName + "-pom.xml"));
 
         // Validate that the schema file is created.
-        File resultFile = new File(generationLocation,testCaseName + "/TestClass-schema.json");
+        File resultFile = new File(generationLocation, testCaseName + "/TestClass-schema.json");
         Assertions.assertTrue(resultFile.exists());
         resultFile.deleteOnExit();
 
         // Validate that is the same as the reference
         File referenceFile = new File(testCaseLocation + "/" + testCaseName + "-reference.json");
         Assertions.assertTrue(referenceFile.exists());
-        Assertions.assertTrue(FileUtils.contentEquals(resultFile, referenceFile),
+        Assertions.assertTrue(FileUtils.contentEqualsIgnoreEOL(resultFile, referenceFile, CHARSET_NAME),
                 "Generated schema for " + testCaseName + " is not equal to the expected reference.");
     }
 
@@ -138,12 +141,12 @@ public class SchemaGeneratorMojoTest extends AbstractMojoTestCase{
         // Validate that they are the same as the reference
         File referenceFileA = new File(testCaseLocation + "/" + "TestClassA-reference.json");
         Assertions.assertTrue(referenceFileA.exists());
-        Assertions.assertTrue(FileUtils.contentEquals(resultFileA, referenceFileA),
+        Assertions.assertTrue(FileUtils.contentEqualsIgnoreEOL(resultFileA, referenceFileA, CHARSET_NAME),
                 "Generated schema for TestClassA is not equal to the expected reference.");
 
         File referenceFileB = new File(testCaseLocation + "/" + "TestClassB-reference.json");
         Assertions.assertTrue(referenceFileB.exists());
-        Assertions.assertTrue(FileUtils.contentEquals(resultFileB, referenceFileB),
+        Assertions.assertTrue(FileUtils.contentEqualsIgnoreEOL(resultFileB, referenceFileB, CHARSET_NAME),
                 "Generated schema for TestClassB is not equal to the expected reference.");
     }
 
@@ -174,17 +177,17 @@ public class SchemaGeneratorMojoTest extends AbstractMojoTestCase{
         // Validate that they are the same as the reference
         File referenceFileA = new File(testCaseLocation + "/" + "TestClassA-reference.json");
         Assertions.assertTrue(referenceFileA.exists());
-        Assertions.assertTrue(FileUtils.contentEquals(resultFileA, referenceFileA),
+        Assertions.assertTrue(FileUtils.contentEqualsIgnoreEOL(resultFileA, referenceFileA, CHARSET_NAME),
                 "Generated schema for TestClassA is not equal to the expected reference.");
 
         File referenceFileB = new File(testCaseLocation + "/" + "TestClassB-reference.json");
         Assertions.assertTrue(referenceFileB.exists());
-        Assertions.assertTrue(FileUtils.contentEquals(resultFileB, referenceFileB),
+        Assertions.assertTrue(FileUtils.contentEqualsIgnoreEOL(resultFileB, referenceFileB, CHARSET_NAME),
                 "Generated schema for TestClassB is not equal to the expected reference.");
 
         File referenceFileC = new File(testCaseLocation + "/" + "TestClassC-reference.json");
         Assertions.assertTrue(referenceFileC.exists());
-        Assertions.assertTrue(FileUtils.contentEquals(resultFileC, referenceFileC),
+        Assertions.assertTrue(FileUtils.contentEqualsIgnoreEOL(resultFileC, referenceFileC, CHARSET_NAME),
                 "Generated schema for TestClassC is not equal to the expected reference.");
     }
 
@@ -212,12 +215,12 @@ public class SchemaGeneratorMojoTest extends AbstractMojoTestCase{
         // Validate that they are the same as the reference
         File referenceFileA = new File(testCaseLocation + "/" + "TestClassA-reference.json");
         Assertions.assertTrue(referenceFileA.exists());
-        Assertions.assertTrue(FileUtils.contentEquals(resultFileA, referenceFileA),
+        Assertions.assertTrue(FileUtils.contentEqualsIgnoreEOL(resultFileA, referenceFileA, CHARSET_NAME),
                 "Generated schema for TestClassA is not equal to the expected reference.");
 
         File referenceFileB = new File(testCaseLocation + "/" + "TestClassB-reference.json");
         Assertions.assertTrue(referenceFileB.exists());
-        Assertions.assertTrue(FileUtils.contentEquals(resultFileB, referenceFileB),
+        Assertions.assertTrue(FileUtils.contentEqualsIgnoreEOL(resultFileB, referenceFileB, CHARSET_NAME),
                 "Generated schema for TestClassB is not equal to the expected reference.");
     }
 

--- a/jsonschema-maven-plugin/src/test/resources/error-test-cases/ClassNotFound-dontFailIfNoClassesMatch-pom.xml
+++ b/jsonschema-maven-plugin/src/test/resources/error-test-cases/ClassNotFound-dontFailIfNoClassesMatch-pom.xml
@@ -1,0 +1,14 @@
+<project>
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>com.github.victools</groupId>
+                <artifactId>jsonschema-maven-plugin</artifactId>
+                <configuration>
+                    <classNames>com.github.victools.jsonschema.plugin.maven.UnknownTestClass</classNames>
+                    <failIfNoClassesMatch>false</failIfNoClassesMatch>
+                </configuration>
+            </plugin>
+        </plugins>
+    </build>
+</project>

--- a/slate-docs/source/includes/_maven-plugin.md
+++ b/slate-docs/source/includes/_maven-plugin.md
@@ -24,6 +24,15 @@ There are a number of basic configuration options as well as the possibility to 
 
 The designated types can be mentioned separately (with dots as package separator) or in the form of glob patterns (with `/` as package separator) in `<classNames>` and/or included as part of their packages through `<packageNames>`. Through `<excludeClassNames>` you can further narrow down the type selection.
 
+```xml
+<configuration>
+    ...
+    <failIfNoClassesMatch>false</failIfNoClassesMatch>
+</configuration>
+```
+
+By default, the plugin aborts if the glob pattern does not match any class. If this is not desired, the `<failIfNoClassesMatch>` property can be set to `false`.
+
 ## Basic Configuration Options
 There are some additional parameters available in the plugin `<configuration>`:
 

